src/librustc_middle/mir/visit.rs

```diff
@@ -413,16 +407,10 @@ macro_rules! make_mir_visitor {
 
             fn super_terminator(&mut self,
                                 terminator: &$($mutability)? Terminator<'tcx>,
-                                location: Location) {
+                                source_location: Location) {
                 let Terminator { source_info, kind } = terminator;
 
                 self.visit_source_info(source_info);
```

Comment:
Note that `visit_terminator_kind` + `visit_source_info` (or `visit_span`) could've been used to statefully track the current `Span`, so if any of the visitors you changed override `visit_source_info` / `visit_span`, there's an accidental behavioral change.

Although maybe we should get the `Span` from `Location` (which is propagated much deeper) nowadays, it'd be overall cleaner.

Reply:
I checked for `visit_source_info` but did not know about `visit_span`, so I will check again.

Reply:
Of the visitors that I patched, none override `visit_span`.